### PR TITLE
PE-5119 POC- Superset Roles

### PR DESCRIPTION
**Pull request by knowvishnu**

### CODE REVIEW

File: superset/security.py [hunk 1]
```diff
@@ -139,6 +139,8 @@ class SupersetSecurityManager(SecurityManager):
 
     GAMMA_ACCESSIBLE_PERMS = {"all_datasource_access"}
 
+    PEAK_USER_PERM = {"all_datasource_access"} | ALPHA_ONLY_VIEW_MENUS
```
Comment (ritikadhyawala) [inline superset/security.py]: not using anywhere

File: superset/security.py [hunk 2]
```diff
@@ -743,6 +746,24 @@ def _is_granter_pvm(self, pvm: PermissionModelView) -> bool:
...
+        """
+
+        return (
+               pvm.view_menu.name
+               in {"SQL Lab", "SQL Editor", "Query Search", "Saved Queries"}
+               or pvm.permission.name
+               in {
+                   "can_sql_json",
+                   "can_csv",
+                   "can_search_queries",
```
Comment (ritikadhyawala) [inline superset/security.py]: should have moved this array outside as PEAKUSER_PERM

File: superset/security.py [hunk 3]
```diff
@@ -743,6 +746,24 @@ def _is_granter_pvm(self, pvm: PermissionModelView) -> bool:
...
         return pvm.permission.name in {"can_override_role_permissions", "can_approve"}
 
+    def _is_peak_user_pvm(self, pvm: PermissionModelView) -> bool:
+        """
+        Return True user have menu access of sql lab
+        """
+
+        return (
+               pvm.view_menu.name
+               in {"SQL Lab", "SQL Editor", "Query Search", "Saved Queries"}
```
Comment (ritikadhyawala) [inline superset/security.py]: For POC, try to disable SQL Lab and check if a peakuser is unable to open SQL Lab. 
Also check if the view itself is hidden if the permisson is not given.

File: superset/custom_auth.py [hunk 4]
```diff
@@ -39,12 +39,18 @@ def login(self):
                     }))['context']
                 if not auth_response['tenant'] == environ['TENANT']:
                     raise Exception('Tenant mismatch in token')
-                if auth_response['role'] in ['tenantManager', 'tenantAdmin']:
+                # keep tenantManagerRole and tenantAdmin role in secret manager or anytwhere that we don't have change dashboard
+                if auth_response['role'] == 'tenantManager': 
                     user = 'admin'
+                elif auth_response['role'] == 'tenantAdmin':
+                    user = 'peakuser'
```
Comment (knowvishnu) [inline superset/custom_auth.py]:
```suggestion
                    user = 'admin'
```

**Pull request closed without merging**